### ISSUE TITLE
Multiple extract-text-webpack-plugin being pulled into config when both processing JS and (S)CSS.

### ISSUE THREAD

**Issue by rbvea** [JeffreyWay/laravel-mix#1737]
- Laravel Mix Version: 2.1.11
- Node Version (`node -v`): 8.9.4
- NPM Version (`npm -v`): 5.6.0
- OS: OSX

### Description:

It appears that there are multiple ExtractTextPlugin's being pushed to the plugins when running js and a preprocessor. A specific problem is using another plugin that depends on ExtractTextPlugin, like purgecss-webpack-plugin.  It seems that it runs purgecss through the css only for the text to be extracted again, causing the css treeshaking to be undone.

### Steps To Reproduce:

I've started up a [repro repo](https://github.com/rbvea/laravel-mix-repro) that demonstrates the issue.

Here's the mix file

```js
const mix = require("laravel-mix");
const purgecssWebpackPlugin = require("purgecss-webpack-plugin");
require("laravel-mix-tailwind");
const glob = require("glob");

if (!process.env.OPTIMIZE_CSS) {
  mix.js("./index.js", "./dist/app.js");
}

class TailwindExtractor {
  static extract(content) {
    return content.match(/[A-Za-z0-9-_:\/]+/g) || [];
  }
}

mix
  .sass("./style.scss", "./dist/style.css")
  .tailwind()
  .webpackConfig({
    plugins: [
      new purgecssWebpackPlugin({
        paths: [].concat(glob.sync(__dirname + "/*.html")),
        extractors: [
          {
            extractor: TailwindExtractor,
            extensions: ["html"]
          }
        ]
      })
    ]
  });
```

When you run `npm run production` the css file is 311kb large despite the purgecss plugin being intact.

When you run `OPTIMIZE_CSS=1 npm run production` the css file is 1.36kb.

**Comment by rbvea** [comment]
Also note that `mix.options({extractVueStyles: false})` does not solve this.

**Comment by rbvea** [comment]
Looks like this was fixed as of v2.1.12 so I'm closing.  Thanks!
